### PR TITLE
[1.6] fix REPLACE w/o PARTIAL

### LINKED ISSUE #647: REPLACE (without PARTIAL) keeps old fields
Expected lastName to be cleared when I run REPLACE without PARTIAL. did I miss something? This happens on both TEXT/NUMERIC and whether I use or don't use NOSTEM/SORTABLE.

```
127.0.0.1:6379> FT.CREATE foo STOPWORDS 0 SCHEMA firstName TEXT NOSTEM SORTABLE lastName TEXT NOSTEM SORTABLE
OK
127.0.0.1:6379> FT.ADD foo 1 1 FIELDS firstName John lastName Doe
OK
127.0.0.1:6379> FT.ADD foo 1 1 REPLACE FIELDS firstName John2
OK
127.0.0.1:6379> FT.GET foo 1
1) "firstName"
2) "John2"
3) "lastName"
4) "Doe"
```

### LINKED ISSUE #1193: FT.ADD ... REPLACE without all fields leaves unindexed fields
If you create a schema with multiple fields then later use `FT.ADD .. REPLACE` but do not specify all the indexed fields, the document will still contain the field yet it won't be present in the index. Tested with 1.4.x and 1.6.x. Note `age` below:

```
> FT.CREATE myppl SCHEMA first TEXT last TEXT age NUMERIC
OK
> FT.ADD myppl instructor 1 FIELDS first Kyle last Davis age 39 country Canada
OK
> FT.SEARCH myppl "@age:[39 39]"
1) (integer) 1
2) "instructor"
3) 1) first
   2) "Kyle"
   3) last
   4) "Davis"
   5) age
   6) "39"
   7) country
   8) "Canada"
> FT.ADD myppl instructor 1 REPLACE FIELDS first Loris last Cro
OK
> FT.GET myppl instructor
1) "first"
2) "Loris"
3) "last"
4) "Cro"
5) "age"
6) "39"
7) "country"
8) "Canada"
> FT.SEARCH myppl "@age:[39 39]"
1) (integer) 0
```

It's unusual behaviour but it seems like it's doing basically a `FT.DEL` without DD, then doing an `FT.ADD` with the specified fields only (which is makes sense since the un-indexed field `country` is also persisting).

### PR DESCRIPTION
**Pull request by ashtul**
fixes #647
fixes #1193

Failures are unrelated to this PR and will be fixed.

### CODE REVIEW

File: src/document_add.c [hunk 1]
```diff
@@ -168,6 +165,20 @@ int RS_AddDocument(RedisSearchCtx *sctx, RedisModuleString *name, const AddDocum
...
 
+  if (!exists) {
+    // If the document does not exist, remove replace/partial settings
+    addOptions &= ~(DOCUMENT_ADD_REPLACE | DOCUMENT_ADD_PARTIAL);
+  } else if (!(addOptions & DOCUMENT_ADD_PARTIAL)) {
+    // delete old decument if REPLACE without PARTIAL
+    RedisModuleKey *dk = RedisModule_OpenKey(sctx->redisCtx, name, REDISMODULE_WRITE);
+    if (dk) {
+      if (RedisModule_KeyType(dk) == REDISMODULE_KEYTYPE_HASH) {
+        RedisModule_DeleteKey(dk);
```
Comment (rafie) [inline src/document_add.c:176]: Do we need to handle deletion like we do in `DeleteCommand()` i.e. with `RedisModule_Replicate()`?

File: src/pytest/test_issues.py [hunk 2]
```diff
@@ -32,3 +32,12 @@ def testMultiSortby(env):
   #TODO: allow multiple sortby steps
   #env.expect('ft.search idx foo nocontent sortby t1 sortby t3').equal(sortby_t1)
   #env.expect('ft.search idx foo nocontent sortby t2 sortby t3').equal(sortby_t2)
+
+def testReplaceIssue(env):
+  # Replace does not remove old fields that are not included in FT.ADD REPLACE command
+  env.cmd('FT.CREATE', 'idx', 'SCHEMA', 't1', 'TEXT', 'SORTABLE', 't2', 'TEXT', 'SORTABLE')
+  env.cmd('FT.ADD', 'idx', 'doc', '1', 'FIELDS', 't1', 'foo', 't2', 'bar')
+  env.expect('HGETALL', 'doc').equal({'t2': 'bar', 't1': 'foo'})
```
Comment (oshadmi) [inline src/pytest/test_issues.py:40]: Suggest to also test `PARTIAL`, before testing `REPLACE` without `PARTIAL`, e.g.,
```
env.cmd('FT.ADD', 'idx', 'doc', '1', 'REPLACE', ,'PARTIAL', 'FIELDS', 't1', 'foz')
env.expect('HGETALL', 'doc').equal({'t2': 'bar', 't1': 'foz'})
```

File: src/document_add.c (hunk 1 above)
Comment (ashtul) [reply src/document_add.c:176]: No, since the `FT.ADD` command calls `RedisModule_Replicate` at the end of execution.

**Comment by ashtul** [comment]
RediSearch 1.6 does not modify the hash and therefore this functionality would not be supported.
To do a full replacement, please delete the document manually prior to the `FT.ADD` command.

**Comment by ashtul** [comment]
Replaced by PR #2018 which adds `REPLACE_DELETE_HASH_FIELD` module parameter.

**Pull request closed without merging**